src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala

```diff
@@ -228,6 +232,7 @@ class FastSync(
       (startBlock to ((startBlock - blocksToDiscard) max 1) by -1).foreach { n =>
         blockchain.getBlockHeaderByNumber(n).foreach { headerToRemove =>
           blockchain.removeBlock(headerToRemove.hash, withState = false)
+          blockSyncCache.invalidate(headerToRemove.hash)
```

Comment:
I think you can safely clear all cache during block discard (it is happening only after validation fail and then we go back by 2048 blocks, or when fastsync finishes.)

Reply:
Ok